py_bind/optv/image_processing.pyx

```diff
@@ -24,38 +23,38 @@ def preprocess_image(np.ndarray[ndim=2, dtype=np.uint8_t] input,
     filter_file - path to a text file containing the filter matrix to be
         used in case ```filter_hp == 2```. One line per row, white-space 
         separated columns.
-    numpy.ndarray output - result numpy 2d array representing the source 
+    numpy.ndarray output_img - result numpy 2d array representing the source 
         image to filter. Same size as img.
     
     Returns:
     numpy.ndarray representing the result image.
     '''
 
     # check arrays dimensions
-    if input.ndim != 2:
-        raise TypeError("Input matrix must be two-dimensional")
-    if output != None and (input.shape[0] != output.shape[0] or
-                         input.shape[1] != output.shape[1]):
+    if input_img.ndim != 2:
```

Comment:
Should be 'is None', not != None. We get a future warning now, that you should have seen when running the tests. This is something I already fixed in other bindings you wrote, and I'll fix it now before submitting upstream, but please pay attention to warnings.